Add support for MIMXRT1170_EVK

s-bruce13 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->
Adding Support for the MIMXRT1170_EVK

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->



### Documentation <!-- Required -->

<!-- 
None
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [X] Tests / results supplied as part of this PR
    tests-mbed_hal-common_ticker*,tests-mbed_hal-us_ticker*
    tests-mbed_hal-common_ticker*,tests-mbed_hal-lp_ticker*
    build successfully
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

s-bruce13 committed:
Added MIMXRT1170_EVK target, UART and Timer

s-bruce13 committed:
Added folder for MIMXRT1170, placed specific files for RT1050 in its own folder and generic in IMX folder

s-bruce13 committed:
Correct serial api file for RT1170

s-bruce13 committed:
Merge branch 'master' into master

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/CMakeLists.txt:
```diff
@@ -22,5 +23,3 @@ target_sources(mbed-mcuxpresso-mcus
         middleware/osa/fsl_os_abstraction_mbed.c
 )
-
-target_link_libraries(mbed-mcuxpresso-mcus INTERFACE mbed-nxp)
```
this should stay

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/CMakeLists.txt:
```diff
@@ -0,0 +1,45 @@
+        drivers/fsl_dcdc.c
+        drivers/fsl_flexspi_nor_boot.c
+        drivers/fsl_gpio.c
+	drivers/fsl_lpuart.c
```
please fix alignment

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/CMakeLists.txt:
```diff
@@ -0,0 +1,45 @@
+    INTERFACE
+        mbed-mcuxpresso-mcus
+        mbed-evk
+	mbed-imx
```
also here

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/TARGET_EVK/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+    INTERFACE
+        PeripheralPins.c
+        fsl_clock_config.c
+	pinmap.c
```
alignment

0xc0170 commented on targets/targets.json:
```diff
@@ -4323,6 +4323,42 @@
         }
     },
+ "MIMXRT1170_EVK": {
+    "supported_form_factors": [
```
align to 4 spaces

0xc0170 reviewed (dismissed):
Please describe your changes in the first section in your first comment (`Summary of changes`).

How was this tested, can you paste logs to confirm this run all tests?

s-bruce13 committed:
Update CMakeLists Alignment

s-bruce13 committed:
Update CMakeLists.txt from review

s-bruce13 committed:
Update CMakeLists

s-bruce13 committed:
Alignment update on target

0xc0170 commented on targets/targets.json:
```diff
@@ -4323,6 +4323,42 @@
+            "SERIAL",
+	        "LPTICKER"
+        ],
+        "device_name": "MIMXRT1170"
```
travis reports an error with

`AssertionError: Target MIMXRT1170_EVK contains invalid device_name MIMXRT1170`. You can either update the index.json from the cmsis pack (https://os.mbed.com/docs/mbed-os/v6.8/program-setup/adding-and-configuring-targets.html) for this device or remove this

s-bruce13 replied:
ok fixed it, is there a way to run this travis test? prior to pull request so I can check all of this

0xc0170 replied:
Each stage in Travis matrix use a script or cmd, I usually run the same command locally to reproduce an issue. Often just going through the logs and finding an offender is sufficient.

s-bruce13 replied:
ok sounds good ill run the commands locally to double check myself before. thank you

s-bruce13 committed:
removed device name

s-bruce13 committed:
serial_api change

serial_api is different for rt1050 and rt1170, placed specific files in their respective evk folder

s-bruce13 committed:
fixed mbed test error for serial, added INTERRUPTIN to device_has

hugueskamba commented on targets/targets.json:
```diff
@@ -4323,6 +4323,42 @@
             "network-default-interface-type": "ETHERNET"
         }
     },
+    "MIMXRT1170_EVK": {
```
Please override `supported_c_libs` to indicate only support for `GCC_ARM`.
If you also support the baremetal profile please override `supported_application_profiles`.

s-bruce13 replied:
I added this now

s-bruce13 committed:
UPDATE TO target.json

s-bruce13 committed:
fixed pinmap to right address

s-bruce13 committed:
Fixed Function MUX for LPUART

s-bruce13 committed:
Fixed errors, all mbed test pass

s-bruce13 committed:
added spi

s-bruce13 committed:
added i2c all tests passed

0xc0170 commented on platform/mbed_lib.json:
```diff
@@ -148,7 +148,7 @@
         "minimal-printf-enable-floating-point": {
             "help": "Enable floating point printing when using minimal printf library",
-            "value": false
+            "value": true
```
This change should be a separate pull request.

0xc0170 replied:
Also file 9.0.0 above should be removed (I can't comment on it)

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/CMakeLists.txt:
```diff
@@ -0,0 +1,49 @@
+        drivers/fsl_gpio.c
+        drivers/fsl_lpi2c.c
+        drivers/fsl_lpspi.c
+	    drivers/fsl_lpuart.c
```
can you fix alignment in this file

s-bruce13 replied:
i moved it, but on my end both of these seem aligned.

hugueskamba replied:
Are you using tabs instead of spaces in your editor?

s-bruce13 replied:
to be honest i dont recall I think I use a mix of both depending on how far in the alignment is.... if I open the fil right now I see it aligned, however if I open it online I can see this issue present..

s-bruce13 replied:
actually, if I click edit online the alignment is fine

hugueskamba replied:
Please ensure you use spaces instead of tabs to have correct indentations everywhere.
Thanks

s-bruce13 replied:
ok fixed

0xc0170 commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/TARGET_EVK/CMakeLists.txt:
```diff
@@ -0,0 +1,25 @@
+        PeripheralPins.c
+        pinmap.c
+        serial_api.c
+	    us_ticker.c
```
same here, align

s-bruce13 replied:
same

hugueskamba replied:
Are you using tabs instead of spaces in your editor?

0xc0170 reviewed (dismissed):
Can you please also rebase, PinNames PR landed on master, I would like to see tests run in Travis (needs a rebase)

s-bruce13 committed:
minimal print reverse, delete 9.0.0

s-bruce13 committed:
align?

s-bruce13 committed:
alignment

s-bruce13 committed:
alignment

s-bruce13 committed:
Merge branch 'master' of https://github.com/ARMmbed/mbed-os

s-bruce13 committed:
aligned w/ space

s-bruce13 committed:
aligned with spaces

s-bruce13 committed:
same

hugueskamba commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/CMakeLists.txt:
```diff
@@ -0,0 +1,49 @@
+    INTERFACE
+        mbed-mcuxpresso-mcus
+        mbed-evk
+	    mbed-imx
```
Please fix this.

s-bruce13 committed:
fixed alignment w/ spaces

hugueskamba reviewed (approved):
LGTM assuming you have ran Greentea tests on the hardware.

s-bruce13 committed:
Update targets.json

s-bruce13 committed:
pinname pass generic test

s-bruce13 committed:
generic and arduino pin validated test

s-bruce13 committed:
returned uart pin

s-bruce13 committed:
add comment device pinname

gpsimenos commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1170/TARGET_EVK/PinNames.h:
```diff
@@ -0,0 +1,270 @@
+
+    // USB Pins
+    USBTX = GPIO_AD_24,
+    USBRX = GPIO_AD_25,
```
Please replace USBTX/USBRX with CONSOLE_TX/CONSOLE_RX 😃

s-bruce13 replied:
Thank you, fixed it.

s-bruce13 committed:
add consoletx/rx

s-bruce13 committed:
Merge remote-tracking branch 'upstream/master'

0xc0170 reviewed (dismissed):
@s-bruce13 Looking at the history, can you rebase to fix some commits ("same" for instance could be squashed, and add better description to some commits). As this is a new target, this could be consolidated into couple of commits.

s-bruce13 committed:
Update cmakelist.txt
